compatibility-verifier/compCheck.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# A script that does rolling upgrade of Pinot components
+# from one version to the other given 2 commit hashes. It first builds  
+# Pinot in the 2 given directories and then upgrades in the following order:
+# Controller -> Broker -> Server
+
+declare -a components=("controller" "broker" "server" "zookeeper")
+
+# get usage of the script 
+function usage() {
+  command=$1
+  if [[ "$command" =~ compCheck.sh$ ]] ; then
+    echo "Usage: $command olderCommit newerCommit [workingDir]"
+  elif [[ "$command" =~ checkoutAndBuild$ ]]; then
+    echo "Usage: $command commitHash dirName"
+  elif [[ "$command" =~ startService$ ]]; then
+    echo "Usage: $command serviceName dirName"
+  elif [[ "$command" =~ stopService$ ]]; then
+    echo "Usage: $command serviceName"
+  fi
+  exit 1
+}
+
+# cleanup the temporary directory when exiting the script
+function cleanup() {
+  if [ -n "$tmpDir" ] && [ -d "$tmpDir" ]; then
+    rm -rf "$tmpDir"
+  fi
+}
+
+# cleanp the temporary directory when the bash script exits 
+trap cleanup EXIT
```

Comment:
Please move all the functions to the top and main code below the function defintions